src/Catalyst.Core.Modules.Kvm/KvmModule.cs

```diff
@@ -24,16 +24,28 @@
 using Autofac;
 using Catalyst.Abstractions.Kvm;
 using Nethermind.Core.Specs;
+using Nethermind.Evm;
+using Nethermind.Logging;
+using Nethermind.Store;
 
 namespace Catalyst.Core.Modules.Kvm
 {
     public class KvmModule : Module
     {
         protected override void Load(ContainerBuilder builder)
         {
+            builder.RegisterType<DeltaExecutor>().As<IDeltaExecutor>().SingleInstance();
             builder.RegisterType<KatVirtualMachine>().As<IKvm>().SingleInstance();
             builder.RegisterType<CatalystSpecProvider>().As<ISpecProvider>();
             builder.RegisterType<CatalystGenesisSpec>().As<IReleaseSpec>();
```

Comment:
Think this needs to be: `builder.RegisterInstance(CatalystGenesisSpec.Instance).As<IReleaseSpec>();`

Reply:
Good that you pointed it out. I think there should be no registration at all. ISpecProvider returns genesis spec on genesis delta only.